godot-ffi/src/global.rs

```diff
@@ -131,57 +94,50 @@ impl<T> Global<T> {
 
 // Encapsulate private fields.
 mod global_guard {
-    use std::ops::{Deref, DerefMut};
-    use std::sync::MutexGuard;
-
-    use super::InitState;
+    use std::{
+        cell::OnceCell,
+        ops::{Deref, DerefMut},
+        sync::MutexGuard,
+    };
 
     /// Guard that temporarily gives access to a `Global<T>`'s inner value.
     pub struct GlobalGuard<'a, T> {
-        // Safety invariant: Is `Initialized`.
-        mutex_guard: MutexGuard<'a, InitState<T>>,
+        mutex_guard: MutexGuard<'a, OnceCell<T>>,
     }
 
     impl<'a, T> GlobalGuard<'a, T> {
-        pub(super) fn new(mutex_guard: MutexGuard<'a, InitState<T>>) -> Option<Self> {
-            match &*mutex_guard {
-                InitState::Initialized(_) => Some(Self { mutex_guard }),
+        pub(super) fn new(mutex_guard: MutexGuard<'a, OnceCell<T>>) -> Option<Self> {
+            match mutex_guard.get() {
+                Some(_) => Some(Self { mutex_guard }),
                 _ => None,
             }
```

Comment:
You can use `map` here

Reply:
no

Reply:
could you try to avoid just saying "no" like this? it isnt very helpful most of the time and we need to ask follow up questions anyway to understand more clearly. why doesn't map work?

Reply:
it results in an error because it moves the mutex guard into a FnOnce

Reply:
```rust
error[E0505]: cannot move out of `mutex_guard` because it is borrowed
   --> godot-ffi/src/global.rs:114:35
    |
113 |         pub(super) fn new(mutex_guard: MutexGuard<'a, OnceCell<T>>) -> Option<Self> {
    |                           ----------- binding `mutex_guard` declared here
114 |             mutex_guard.get().map(|_| Self { mutex_guard })
    |             -----------       --- ^^^        ----------- move occurs due to use in closure
    |             |                 |   |
    |             |                 |   move out of `mutex_guard` occurs here
    |             |                 borrow later used by call
    |             borrow of `mutex_guard` occurs here

```

Reply:
I can do this though if you prefer
```rust
mutex_guard.get().is_some().then_some(Self { mutex_guard })
```

Reply:
you see what i have right now is basically an eager map

Reply:
you see what i have right now is basically an eager map

Reply:
Ok, then let's leave the `match`. Can you add a comment mentioning that we can't use `.get().map(...)` due to the guard remaining borrowed?